src/test/util/cliConfig.cpp

```diff
@@ -0,0 +1,322 @@
+// SPDX-License-Identifier: BSD-2-Clause
```

Comment:
Feels like an off-the-shelf solution can be used instead of bringing your own.

Reply:
Less external libraries, the better. Beside, cliConfig class are re-used from Cxbx-Reloaded repository which I had originally written for. This is a revised version allow without dependency on external project such as Unit Test. With ability to use callback to determine which arguments are valid or not. And re-license to BSD-2-Cause instead of GPLv2 or later.

P.S. I had not heard of any "off-the-shelf" solution for cli's arguments.